### PR TITLE
feat:  State channels Hints

### LINKED ISSUE #230: State channels page hints
Add hints to state channels page.

### PR DESCRIPTION
**Pull request by janmichek**
## Description
resolves #230 

sources used:
https://aeternity.com/state-channels
https://github.com/aeternity/state-channel-demo/blob/develop/TUTORIAL.md
https://github.com/aeternity/state-channel-demo

## Checklist:

- [x] Add State Channel hints
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/utils/hints/stateChannelsHints.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export const stateChannelsHints = {
+  stateChannelId: 'Unique identifier of the State Channel.',
+  status: 'State Channel can be Open or Closed. When status is Open the transactions lies outside of the blockchain network.',
+  participants: 'Initiator is an account that created the State Channel via ChannelCreateTx. Responder is an account that interacted with the State Channel as a last one.',
+  onChainUpdates: 'Amount of an off-chain transactions within the State Channel.',
+  locked: 'Currently deposited amount to the State Channel by it\'s participants.',
+  lastKnownRound: '',
```
Comment (janmichek) [inline src/utils/hints/stateChannelsHints.js:7]: Here I was a bit desperate to coming up with a hint. In the sources I went through I haven/t find anything meaningful ./ Can we ignite the discussion in here, please?

File: src/components/AppTooltip.vue [hunk 2]
```diff
@@ -48,7 +48,7 @@ defineProps({
         background: var(--color-midnight);
         color: var(--color-white);
         border: none;
-        word-break: break-all;
+        word-break: break-word;
```
Comment (janmichek) [inline src/components/AppTooltip.vue:51]: This is more readable now in my eyes

File: src/utils/hints/stateChannelsHints.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export const stateChannelsHints = {
+  stateChannelId: 'Unique identifier of the State Channel.',
+  status: 'State Channel can be Open or Closed. When status is Open the transactions lies outside of the blockchain network.',
```
[the remark this answers is not in the document]
Comment (thepiwo) [reply src/utils/hints/stateChannelsHints.js:3]: "When status is Open transactions can be executed inside the State Channel, outside the blockchain network"

File: src/utils/hints/stateChannelsHints.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+  participants: 'Initiator is an account that created the State Channel via ChannelCreateTx. Responder is an account that interacted with the State Channel as a last one.',
```
[the remark this answers is not in the document]
Comment (thepiwo) [reply src/utils/hints/stateChannelsHints.js:4]: responder is the second participant of the channel, not who interacted last, right? cc @marc0olo

**Comment by janmichek** [comment]
Here is some issue with fighting for a space. I will have to fix it
image

File: src/utils/hints/stateChannelsHints.js (hunk 1 above)
Comment (janmichek) [reply src/utils/hints/stateChannelsHints.js]: Thanks for the suggestion, I have used that.

File: src/utils/hints/stateChannelsHints.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+  onChainUpdates: 'Amount of an off-chain transactions within the State Channel.',
```
[the remark this answers is not in the document]
Comment (janmichek) [reply src/utils/hints/stateChannelsHints.js]: removed

**Comment by janmichek** [comment]
> Here is some issue with fighting for a space. I will have to fix it image

The only idea I am coming up with is to shorten some table headers. Giving the hints providing more explanation I just prepared, I think it makes sense.
I propose to shorten `On chain updates` to `Updates`
and `Last Known Round` to `Last Round`

So this final outcome will look like this (not pushed to preview yet). Please consult @thepiwo @michele-franchi @lukeromanowicz 

image

**Comment by lukeromanowicz** [comment]
I think that "known" part might be actually very important in order not to mislead the users, but overall I prefer going down this part over wrapping the question marks to the second line.

My alternative proposal is to put this task on hold until the content width gets expanded as we discussed during the recent workshops.

**Comment by michele-franchi** [comment]
@lukeromanowicz @janmichek for sure we should not display hints on new lines but to completely get rid of the issue we have two options: 
- increase min width of the container
- remove some columns

**Comment by janmichek** [comment]
> @lukeromanowicz @janmichek for sure we should not display hints on new lines but to completely get rid of the issue we have two options:
> 
>     * increase min width of the container
> 
>     * remove some columns
> 
> 
> But maybe It also could be acceptable to make column names shorter and explain with tooltips, if this completely solves the width issue.

Thanks for summing up the options.

1) What column(s) do you suggest to remove?
2) I am still more in favor of shortening labels. The less evil I present now is to shorten `On-chain updates` to `Updates` and `Last Updated` to `Last Update`

It will look like this (still dense) 
image

**Comment by lukeromanowicz** [comment]
During the daily we agreed to remove Last Known Round to keep consistency with a similar state channels table that we use in the Dashboard

**Comment by janmichek** [comment]
> @lukeromanowicz @janmichek for sure we should not display hints on new lines but to completely get rid of the issue we have two options:
> 
>     * increase min width of the container
> 
>     * remove some columns
> 
> 
> But maybe It also could be acceptable to make column names shorter and explain with tooltips, if this completely solves the width issue.

Thanks for summing up the options.

1) What column(s) do you suggest to remove?
2) I am still more in favor of shortening labels. The less evil I present now is to shorten `On-chain updates` to `Updates` and `Last Updated` to `Last Update`

It will look like this (still dense) 
image


> During the daily we agreed to remove Last Known Round to keep consistency with a similar state channels table that we use in the Dashboard

Done

File: src/utils/hints/stateChannelsHints.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+export const stateChannelsHints = {
+  stateChannelId: 'Unique identifier of the State Channel.',
+  status: 'When status is \'Open\' transactions can be executed inside the State Channel, outside the blockchain network',
+  participants: 'Initiator is the account that created the State Channel via ChannelCreateTx.',
+  onChainUpdates: 'Amount of off-chain transactions within the State Channel.',
```
Comment (lukeromanowicz) [inline src/utils/hints/stateChannelsHints.js:5]: that looks weird. I would assume that on-chain updates are... well.. on chain :D Is it correct?

**Review by lukeromanowicz — APPROVED** [review]
Apart from one comment to double check, everything else seems right. Really nice job! For some reason the preview link didn't get generated so here it is:

https://pr-274-aescan.stg.aepps.com/state-channels

File: src/utils/hints/stateChannelsHints.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+  locked: 'Currently deposited amount to the State Channel by it\'s participants.',
```
Comment (michele-franchi) [inline src/utils/hints/stateChannelsHints.js:6]: ...`by its participants.`

File: src/utils/hints/stateChannelsHints.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (janmichek) [reply src/utils/hints/stateChannelsHints.js]: fixed

File: src/utils/hints/stateChannelsHints.js (hunk 7 above)
Comment (janmichek) [reply src/utils/hints/stateChannelsHints.js]: fixed

File: src/utils/hints/stateChannelsHints.js (hunk 6 above)
Comment (janmichek) [reply src/utils/hints/stateChannelsHints.js]: fixed

File: src/utils/hints/stateChannelsHints.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
+export const stateChannelsHints = {
+  stateChannelId: 'Unique identifier of the State Channel.',
+  status: 'When status is "Open" transactions can be executed inside the State Channel, outside the blockchain network.',
+  participants: 'Initiator is the account that created the State Channel via ChannelCreateTx.',
+  onChainUpdates: 'Amount of on-chain transactions within the State Channel.',
+  locked: 'Currently deposited amount to the State Channel by its participants.',
+  lastUpdated: 'Estimated date, time, and block height of the last interaction with the State Channel.',
```
Comment (michele-franchi) [inline src/utils/hints/stateChannelsHints.js:7]: Same for other PRs, could you please adjust it to 
"Block height and estimated date and time of the last interaction with the State Channel."

**Review by michele-franchi — APPROVED** [review]
I left one more suggestion as discussed in other PRs but in general it looks good.

File: src/utils/hints/stateChannelsHints.js (hunk 8 above)
Comment (janmichek) [reply src/utils/hints/stateChannelsHints.js]: I created follow-up on this as it is already used in merged parts https://github.com/aeternity/aescan/issues/307
But I also fixed it here.

**Pull request merged**